src/nbl/builtin/CMakeLists.txt

```diff
@@ -328,3 +322,34 @@ LIST_BUILTIN_RESOURCE(NBL_RESOURCES_TO_EMBED "hlsl/memory_accessor.hlsl")
 LIST_BUILTIN_RESOURCE(NBL_RESOURCES_TO_EMBED "hlsl/enums.hlsl")
 
 ADD_CUSTOM_BUILTIN_RESOURCES(nblBuiltinResourceData NBL_RESOURCES_TO_EMBED "${NBL_ROOT_PATH}/include" "nbl/builtin" "nbl::builtin" "${NBL_ROOT_PATH_BINARY}/include" "${NBL_ROOT_PATH_BINARY}/src" "STATIC" "INTERNAL")
+
+get_filename_component(NBL_SPRIV_INTRINSICS_GEN_INCLUDE_DIR "${CMAKE_CURRENT_BINARY_DIR}/include" ABSOLUTE)
+set(NBL_SPRIV_INTRINSICS_GEN_INCLUDE_DIR "${NBL_SPRIV_INTRINSICS_GEN_INCLUDE_DIR}")
+
+set(NBL_OUTPUT_FILE "${NBL_SPRIV_INTRINSICS_GEN_INCLUDE_DIR}/nbl/builtin/hlsl/spirv_intrinsics/core.hlsl")
+file(MAKE_DIRECTORY "${NBL_SPRIV_INTRINSICS_GEN_INCLUDE_DIR}/nbl/builtin/hlsl/spirv_intrinsics")
+
+set(NBL_GEN_PY "${NBL_ROOT_PATH}/3rdparty/dxc/dxc/external/SPIRV-Headers/tools/hlsl_generator/gen.py")
+set(NBL_COMMAND
+	"${_Python3_EXECUTABLE}"
+	"${NBL_GEN_PY}"
+	"${NBL_OUTPUT_FILE}"
+)
+
+add_custom_command(OUTPUT "${NBL_OUTPUT_FILE}"
+    DEPENDS ${NBL_GEN_PY}
+    COMMAND ${NBL_COMMAND}
+    VERBATIM
+)
+
+add_custom_target(SprivIntrinsics DEPENDS "${NBL_OUTPUT_FILE}")
+
+if(NBL_EMBED_BUILTIN_RESOURCES)
+	LIST_BUILTIN_RESOURCE(NBL_SPRIV_INTRINSICS_GEN_RESOURCES_TO_EMBED "core.hlsl")
+	ADD_CUSTOM_BUILTIN_RESOURCES(sprivIntrinsicsGenBuiltinResourceData NBL_SPRIV_INTRINSICS_GEN_RESOURCES_TO_EMBED "${NBL_SPRIV_INTRINSICS_GEN_INCLUDE_DIR}" "nbl/builtin/hlsl/spirv_intrinsics" "nbl::builtin" "${CMAKE_CURRENT_BINARY_DIR}/include" "${CMAKE_CURRENT_BINARY_DIR}/src" "STATIC" "INTERNAL")
+	add_dependencies(sprivIntrinsicsGenBuiltinResourceData SprivIntrinsics)
+endif()
+
+set(NBL_SPRIV_INTRINSICS_GEN_INCLUDE_DIR
+	"${NBL_SPRIV_INTRINSICS_GEN_INCLUDE_DIR}"
+PARENT_SCOPE)
```

Comment:
you cannot keep it (those are only for `nblBuiltinResourceData`) since you generate your output on fly + you will have different absolute archive entry, you have to follow the device gen target and keep it in separate directory

Reply:
you main issue is you don't add your archive into [`ISystem.cpp`](https://github.com/Devsh-Graphics-Programming/Nabla/blob/master/src/nbl/system/ISystem.cpp#L27) hence you cannot even request this output at runtime